src/components/Map/EditableMapV2.tsx

```diff
@@ -135,52 +140,44 @@ export default function EditableMap({
     });
   }, [readOnlyBoundary]);
 
-  // close popup on click
-  const onPopupClose = useCallback(() => {
-    setPopupInfo((curr) => (curr ? { ...curr, active: false } : null));
-  }, []);
-
   // clear popup state and set geometry as not selected (so rendering shows it as not selected)
   const clearPopupInfo = useCallback(() => {
-    if (!popupInfo) {
-      return;
-    }
-    const { sourceId, id } = popupInfo;
-    mapRef?.current?.setFeatureState({ source: sourceId, id: `${id}` }, { select: false });
-    setPopupInfo(null);
-  }, [mapRef, popupInfo]);
+    setPopupInfo((curr) => {
+      if (curr) {
+        const { sourceId, id } = curr;
+        // clear selection
+        mapRef?.current?.setFeatureState({ source: sourceId, id: `${id}` }, { select: false });
+      }
+      return null;
+    });
+  }, [mapRef]);
 
   // set new popup state and mark geometry as selected (renders as selected)
   const initializePopupInfo = useCallback(
     (info: PopupInfo) => {
-      setPopupInfo((current) => {
-        // cannot call clear popup info to avoid infinite updates
-        if (current) {
-          const { sourceId, id } = current;
-          mapRef?.current?.setFeatureState({ source: sourceId, id: `${id}` }, { select: false });
-        }
-        mapRef?.current?.setFeatureState({ source: info.sourceId, id: `${info.id}` }, { select: true });
-        return info;
-      });
+      clearPopupInfo();
+      // update selection
+      mapRef?.current?.setFeatureState({ source: info.sourceId, id: `${info.id}` }, { select: true });
+      setPopupInfo(info);
     },
-    [mapRef]
+    [clearPopupInfo, mapRef]
   );
 
   // renderer memoized
   const renderedPopup = useMemo<JSX.Element | null>(() => {
     if (!popupInfo || !popupRenderer) {
       return null;
     }
-    return popupRenderer.render(popupInfo.properties, onPopupClose);
-  }, [onPopupClose, popupInfo, popupRenderer]);
+    return popupRenderer.render(popupInfo.properties, clearPopupInfo);
+  }, [clearPopupInfo, popupInfo, popupRenderer]);
 
   // map click to fetch geometry and show a popup at that location
   const onMapClick = useCallback(
     (event: any) => {
       const { lat, lng } = event.lngLat;
 
       // handle overlap with polygon editor clicks
-      if (editMode === 'CreatingBoundary') {
+      if (editMode === 'CreatingBoundary' || overridePopupInfo) {
```

Comment:
we won't show a popup while the existing popup on new polygon creation is still active, this helps remove race conditions for now.